### PR TITLE
Revamp public IS doc

### PR DESCRIPTION
**Pull request by francoisl**
@yuwenmemon Please review / proofread.

For now the text is mostly a copy-paste of the old docs with a few more details, but we can improve over time.

/cc @conorpendergrast in case you want to proofread as well.

### CODE REVIEW

File: source/index.html.md [hunk 1]
```diff
@@ -1,185 +1,1260 @@
...
 
 search: true
 ---
 
 # Introduction
 
-Welcome to the Kittn API! You can use our API to access Kittn API endpoints, which can get information on various cats, kittens, and breeds in our database.
-
-We have language bindings in Shell, Ruby, and Python! You can view code examples in the dark area to the right, and you can switch the programming language of the examples with the tabs in the top right.
+Welcome to the documentation for the Expensify Integration Server. This documentation is designed to give you all the information you need to integrate with Expensify. You will find information on how to programmatically download expense reports data for analysis or insertion into your accounting package, upload employee data to configure or provision accounts for new hires or insert report data into a user’s account from your external system.
```
Comment (yuwenmemon) [inline source/index.html.md]: This is designed to give you all the information you need to integrate with Expensify.

Comment (yuwenmemon) [reply source/index.html.md]: > You will find information on how to programmatically download expense reports data for analysis or insertion into your accounting package, upload employee data to configure or provision accounts for new hires or insert report data into a user’s account from your external system.

Sentence is too long, can you break it up?

File: source/index.html.md [hunk 2]
```diff
@@ -1,185 +1,1260 @@
...
+> API call format
 
-```ruby
-require 'kittn'
+```shell
+curl -X POST 'https://integrations.expensify.com/Integration-Server/ExpensifyIntegrations' \
+    -d 'requestJobDescription={...}'
+```
+
+Every request to the Expensify API follows the same pattern: a JSON payload that identifies the job to execute is passed, as a parameter called `requestJobDescription`. Additionally, other parameters may be needed based on the type of job.
```
Comment (yuwenmemon) [inline source/index.html.md]: a JSON payload that identifies the job to execute is passed as a parameter called `requestJobDescription`
(no comma)

File: source/index.html.md [hunk 3]
```diff
@@ -1,185 +1,1260 @@
...
+            }
+        },
+        "outputSettings":{
+            "fileExtension":"csv"
+        }
+    }'
+    --data-urlencode 'template@expensify_template.ftl'
 ```
 
+> - Export at most 10 Approved and Reimbursed reports generated between 2016-01-01 and 2016-02-01 to an xls file. The exported reports are then marked as exported with the label "Expensify Report" (markAsExported action), and an email with a link to the output is sent. Reports that have already been marked with the label "Expensify Report" are ignored (markedAsExported filter)
```
Comment (yuwenmemon) [inline source/index.html.md]: "Expensify Export"

Comment (yuwenmemon) [reply source/index.html.md]: to an xls file named `myExport`

File: source/index.html.md [hunk 4]
```diff
@@ -1,185 +1,1260 @@
...
+            {"actionName":"markAsExported","label":"Expensify Export"},
+            {"actionName":"email","recipients":"manager@domain.com,finances@domain.com", "message":"Report is ready."}
+        ]
+    }'
+    --data-urlencode 'template@expensify_template.ftl'
 ```
 
-```javascript
-const kittn = require('kittn');
+> - expensify_template.ftl sample
```
Comment (yuwenmemon) [inline source/index.html.md]: Maybe lets link to the freemarker doc https://integrations.expensify.com/Integration-Server/doc/export_report_template.html

File: source/index.html.md [hunk 5]
```diff
@@ -1,185 +1,1260 @@
...
+        ${reportNumber},<#t>
+        ${expenseNumber}<#lt>
+        <#assign expenseNumber = expenseNumber + 1>
+    </#list>
+    <#assign reportNumber = reportNumber + 1>
+</#list>
 ```
 
-> Make sure to replace `meowmeowmeow` with your API key.
+Export expense or report data for analysis or insertion into your accounting package, in a configurable format.
```
Comment (yuwenmemon) [inline source/index.html.md]: Export expense or report data in a configurable format for analysis or insertion into your accounting package.

File: source/index.html.md [hunk 6]
```diff
@@ -1,185 +1,1260 @@
...
+
+### `requestJobDescription` format
+
+Name | Format | Valid values | Description
+--------- | --------- | --------- | ---------
+type | String | file | |
+onReceive | JSON object | `{"immediateResponse":["returnRandomFileName"]}` | Returns the name of the generated report.
```
Comment (yuwenmemon) [inline source/index.html.md]: If this is the only valid value I think we should explain in the Description

File: source/index.html.md [hunk 7]
```diff
@@ -1,185 +1,1260 @@
...
+limit | String | Any integer, as a string | Maximum number of reports to export. |
+employeeEmail | String | A valid email address<br>**Note:**<br> * The usage of this parameter is restricted to certain domains<br>* If this parameter is used, reports in the OPEN status cannot be exported  | The reports will be exported from that account.
+
+- `inputSettings.filters`
+
+Name | Format | Valid values | Description
+--------- | --------- | --------- | ---------
+reportIDList | String, <br>*Optional* |  | Comma-separated list of report IDs to be exported.
+policyIDList | String, <br>*Optional* |  | Comma-separated list of policy IDs the exported reports must be under.
+startDate | Date, <br>*Required if `reportIDList` is not specified* | yyyy-mm-dd formatted date | Filters all reports submitted or created before the given date, whichever occurred last (inclusive).
```
Comment (yuwenmemon) [inline source/index.html.md]: Filters out, same for below

File: source/index.html.md [hunk 8]
```diff
@@ -1,185 +1,1260 @@
...
+Name | Format | Valid values | Description
+-------- | --------- | ---------------- | ---------
+fileExtension | String | One or multiple of "csv", "xls", "xlsx", "txt",  "pdf", "json", "xml" | Specifies the format of the generated report. <br />*Note*: if the "pdf" option is chosen, one PDF file will be generated for each report exported.
+**Optional elements** |
+fileBasename | String | Any valid file base name | The name of the generated file(s) will start with this value, and a random part will be added to make each filename globally unique. If not specified, the default value `export` is used. |
+includeFullPageReceiptsPdf | Boolean | `true`, `false` | Specifies whether generated PDFs should include full page receipts. This parameter is used only if `fileExtension` contains `pdf`. |
+spreadsheetFilename | String | Name of a workbook template stored for you on Expensify. (e.g. "testWorkbook.xlsx")| Specifies workbook template that report data will be written to. <br />*Note*: if this is provided, then the only file made available to export will be an xlsx file. All other file extensions provided in `fileExtension` will be ignored.
```
Comment (yuwenmemon) [inline source/index.html.md]: Can we mark this as `enterprise` somehow or something?

File: source/index.html.md [hunk 9]
```diff
@@ -1,185 +1,1260 @@
...
+fields | JSON Object | A series of JSON objects whose key is the name of the report field to modify, and the value is the value to set. The key needs to have all non-alphanumerical characters replaced with underscores (`_`). | Values for custom fields on the specified `policyID`.
+
+- `inputSettings.expenses`
+
+Name | Format | Valid values | Description
+-------- | --------- | ---------------- | ---------
+merchant | String |  | The title of the report that will be created.
+currency | String | Three-letter currency code of the transaction. | The currency in which the transaction was made.
+date | Date | yyyy-mm-dd formatted date | The date the expense was made.
+amount | Integer |  | The amount of the transaction, in pennies.
```
Comment (yuwenmemon) [inline source/index.html.md]: cents, not pennies

File: source/index.html.md [hunk 10]
```diff
@@ -1,185 +1,1260 @@
...
+                            "glCode": "Tag 4 GL Code"
+                        }
+                    ]
+                }
+            ]
+        }
+    }'
+```
+
+> - Dependent multi-level tags, with GL Codes and tag level names, passed in CSV.
```
Comment (yuwenmemon) [inline source/index.html.md]: Dependent multi-level tags, with GL Codes and tag level names. These must be passed via CSV.

File: source/index.html.md [hunk 11]
```diff
@@ -1,185 +1,1260 @@
...
+
+> Tag file - save the following data in a file called `tags.csv`, for example.
```
Comment (yuwenmemon) [inline source/index.html.md]: Tag file - In this example, we have the following data in a file called `tags.csv`.

File: source/index.html.md [hunk 12]
```diff
@@ -1,185 +1,1260 @@
...
+### Update tags<a name="update-tags">&nbsp;</a>
+
+Tags can be specified in two ways:
+
+- Directly in the JSON job description
+- As extra parameters of the request
+
+#### Tag levels dependency
+For multi-level tagging, Expensify has the notion of tag level dependency. This means each tag level is dependent on the parent tag level above it.
+For more information and examples about multi-tagging, please refer to our [help site](https://docs.expensify.com/setup-for-admins-and-accountants/advanced-configuration-for-managers-and-accountants/tags).
```
Comment (yuwenmemon) [inline source/index.html.md:1000]: mention that it would be in csv format that way

File: source/index.html.md [hunk 13]
```diff
@@ -1,185 +1,1260 @@
...
+
+Name | Format | Description
+-------- | --------- | ---------
+name | String | The name of the tag.
+**Optional elements** |
+enabled | Boolean | Whether the tag is enabled or not. Default value is `true`.<br>*Note:* When multi-level tagging is used, this value is ignored and is consider `true`.
+glCode | String | The GL Code associated to the tag.
+
+
+#### Passing tag data from plain text as additional request parameters (dependent or independent tag levels)
```
Comment (yuwenmemon) [inline source/index.html.md:1030]: considered

File: source/index.html.md [hunk 14]
```diff
@@ -1,185 +1,1260 @@
...
+
+- `tags.config`
+
+Name | Format | Valid values | Description
+-------- | --------- | ---------------- | ---------
+dependency | Boolean | true, false | Whether tag level are dependent. |
+glCodes | Boolean | true, false | Whether adjacent columns in the tag file(s) contain GL Codes.
+header | Boolean | true, false | Whether the first line of the file contains the names of the tag levels.
+setRequired | Boolean | true, false | If set to `true`, users will be required to tag each expenses under that category.
+fileType | String | "cvs" or "tsv" | Format of the tag data.
```
Comment (yuwenmemon) [inline source/index.html.md:1050]: levels

File: source/index.html.md [hunk 15]
```diff
@@ -1,185 +1,1260 @@
...
+
+## Employee updater
+
```
Comment (yuwenmemon) [inline source/index.html.md:1053]: expense

File: source/index.html.md [hunk 16]
```diff
@@ -1,185 +1,1260 @@
...
+    "responseMessage": "Column EmployeeEmail not found in CSV header",
+    "responseCode": 500
+}
+```
+
+Lets you manage employees on an Expensify policy. Use this to upload employee data in order to configure or provision accounts for new hires.
+
+The employee data must be passed in the request parameter `data`.
+
+- `inputSettings`
```
Comment (yuwenmemon) [inline source/index.html.md:1115]: Use this to configure or provision accounts for new hires, as well as to update existing employee data

File: source/index.html.md [hunk 17]
```diff
@@ -1,185 +1,1260 @@
...
+fileType | String | "csv" | The format of the file that contains the employee data. Only CSV is supported at this point.
+policyID | String | Any valid Expensify policy ID, owned or shared with the user. | The policy to add the employees to.
+
+- `data` request argument
+
+CSV file containing the employee data to update. The first line of the file lists the columns that will exist. The order of each attribute does not matter.
+
+Name | Format | Valid values | Description
+-------- | --------- | ---------------- | ---------
+**Required CSV columns** |
```
Comment (yuwenmemon) [inline source/index.html.md:1129]: The order of the columns does not matter

File: source/index.html.md [hunk 18]
```diff
@@ -1,185 +1,1260 @@
...
+EmployeeEmail | String | Any valid email address | The email address of the employee to update or create.
+ManagerEmail | String | Any valid email address | The email address of the manager of that user (corresponds to the column `Submits To` on Expensify).
+Admin | Boolean | true, false | Determines whether that user has administrator privileges.
+**Optional CSV columns** |
+EmployeeUserId | String | | The User ID of the employee.
+EmployeePayrollId | String | | The Payroll ID of the employee.
```
Comment (yuwenmemon) [inline source/index.html.md:1135]: The email address the employee's manager

File: source/index.html.md [hunk 19]
```diff
@@ -1,185 +1,1260 @@
...
+type | String | "reportStatus" | Specifies to the job that it has to update the status of a list of reports.
+status | String | "REIMBURSED" | The status to change the reports to. At the moment, only Reimbursed is supported.<br>**Note: at the moment, the only supported action is to mark Approved reports as Reimbursed.**<br>**Only reports in the `Approved` status can be updated to `Reimbursed`. All other reports will be ignored.** |
+filters | JSON object | See inputSettings filters |
 
-### URL Parameters
 
-Parameter | Description
---------- | -----------
-ID | The ID of the kitten to retrieve
+- `inputSettings.filters`
```
Comment (yuwenmemon) [inline source/index.html.md:1262]: You state the face that you can only mark as reimbursed is supported twice.

**Comment by francoisl** [comment]
Updated

File: source/index.html.md (hunk 19 above)
Comment (francoisl) [reply source/index.html.md:1262]: Removed the other one

File: source/index.html.md [hunk 20]
```diff
@@ -1,185 +1,1260 @@
...
 
 search: true
 ---
 
 # Introduction
 
-Welcome to the Kittn API! You can use our API to access Kittn API endpoints, which can get information on various cats, kittens, and breeds in our database.
-
-We have language bindings in Shell, Ruby, and Python! You can view code examples in the dark area to the right, and you can switch the programming language of the examples with the tabs in the top right.
+Welcome to the documentation for the Expensify Integration Server. This is designed to give you all the information you need to integrate with Expensify. You will find information on how to programmatically download expense reports data for analysis or insertion into your accounting package, provision or update accounts for new hires, and much more.
```
Comment (yuwenmemon) [inline source/index.html.md]: You will find information on how to programmatically download expense report data for analysis or insertion into your accounting package, provision accounts for new hires, and much more.

File: source/index.html.md [hunk 21]
```diff
@@ -1,185 +1,1260 @@
...
+For more information, please contact us at <help@expensify.com>. Enjoy!
 
 # Authentication
 
-> To authorize, use this code:
+To use the API, you will need to generate API credentials.
+
+1. Create an Expensify account at <https://www.expensify.com/>
+2. Go to <https://www.expensify.com/tools/integrations/>
+3. A pair of credentials `partnerUserID` and `partnerUserSecret` will be generated and shown on the page.
```
Comment (yuwenmemon) [inline source/index.html.md]: A pair of credentials: `partnerUserID` and `partnerUserSecret` will be generated and shown on the page.

File: source/index.html.md [hunk 22]
```diff
@@ -1,185 +1,1260 @@
...
 ```
 
-```python
-import kittn
+For every request, the `requestJobDescription` JSON parameter will need to contain the following data:
+
+Parameter | Type | Description
+--------- | ------- | -----------
+type | String | The type of job to execute
+credentials | JSON object | An object containing two key/values: `partnerUserID` and `partnerUserSecret`, used to authenticate you
```
Comment (yuwenmemon) [inline source/index.html.md]: An object containing two key/values used to authenticate you: `partnerUserID` and `partnerUserSecret`

File: source/index.html.md [hunk 23]
```diff
@@ -1,185 +1,1260 @@
...
+
+
+# Export
+
+## Report Exporter
+
+> The following examples assume the template is stored in the file 'expensify_template.ftl'
 
-api = kittn.authorize('meowmeowmeow')
+> - Export report by IDs, to a CSV file
```
Comment (yuwenmemon) [inline source/index.html.md]: Export reports by ID,

File: source/index.html.md [hunk 24]
```diff
@@ -1,185 +1,1260 @@
...
+            }
+        },
+        "outputSettings":{
+            "fileExtension":"csv"
+        }
+    }'
+    --data-urlencode 'template@expensify_template.ftl'
 ```
 
+> - Export at most 10 Approved and Reimbursed reports generated between 2016-01-01 and 2016-02-01 to an xlsx file starting with `myExport`. The exported reports are then marked as exported with the label "Expensify Export" (`markAsExported` action), and an email with a link to the output is sent. Reports that have already been marked with the label "Expensify Export" are ignored (`markedAsExported` filter)
```
Comment (yuwenmemon) [inline source/index.html.md]: Export at most 10 Approved and Reimbursed reports generated between 2016-01-01 and 2016-02-01 to an xlsx file named `myExport.xlsx`.

Comment (yuwenmemon) [reply source/index.html.md]: and an email linking to the output is sent

File: source/index.html.md [hunk 25]
```diff
@@ -1,185 +1,1260 @@
...
+employeeEmail | String | A valid email address<br>**Note:**<br> * The usage of this parameter is restricted to certain domains<br>* If this parameter is used, reports in the OPEN status cannot be exported  | The reports will be exported from that account.
+
+- `inputSettings.filters`
+
+Name | Format | Valid values | Description
+--------- | --------- | --------- | ---------
+reportIDList | String, <br>*Optional* |  | Comma-separated list of report IDs to be exported.
+policyIDList | String, <br>*Optional* |  | Comma-separated list of policy IDs the exported reports must be under.
+startDate | Date, <br>*Required if `reportIDList` is not specified* | yyyy-mm-dd formatted date | Filters out all reports submitted or created before the given date, whichever occurred last (inclusive).
+endDate | Date, <br> *Optional* | yyyy-mm-dd formatted date | Filters all reports submitted or created after the given date, whichever occurred last (inclusive). |
```
Comment (yuwenmemon) [inline source/index.html.md]: Filters out

File: source/index.html.md [hunk 26]
```diff
@@ -1,185 +1,1260 @@
...
+approvedAfter | Date, <br> *Optional* | yyyy-mm-dd formatted date | Filters out all reports approved before, or on that date. This filter is only used against reports that have been approved. |
+markedAsExported | String,<br>*Optional* | Any string | Filters reports that have already been exported with that label out.
```
Comment (yuwenmemon) [inline source/index.html.md]: Filters out

File: source/index.html.md [hunk 27]
```diff
@@ -1,185 +1,1260 @@
...
+spreadsheetFilename | String | Name of a workbook template stored for you on Expensify. (e.g. "testWorkbook.xlsx") | Specifies workbook template that report data will be written to. <br />*Note*: if this is provided, then the only file made available to export will be an xlsx file. All other file extensions provided in `fileExtension` will be ignored. This is an enterprise feature.
 
-`Authorization: meowmeowmeow`
+- `onFinish` (Optional)
+
+This section can be used to describe actions that need to be performed at the end of the export.
+
+Action Name  | Description
+-------- | ---------
+email | Send an email with a link to the generated report to a list of recipients.
```
Comment (yuwenmemon) [inline source/index.html.md]: Send an email linking to the generated report to a provided list of recipients.

File: source/index.html.md [hunk 28]
```diff
@@ -1,185 +1,1260 @@
...
+                    "amount": 2211
+                }
+            ]
+        }
+    }'
+```
+
+> Response
+
+> - A success response message is comprised of a `responseCode` `200`, the name of the generated report and its ID on Expensify.
```
Comment (yuwenmemon) [inline source/index.html.md]: A success response message is comprised of a `responseCode` `200`, the name of the generated report, and its ID on Expensify.

File: source/index.html.md [hunk 29]
```diff
@@ -1,185 +1,1260 @@
...
+> - Error
+
+```
+{
+    "responseMessage": "Not authorized to authenticate as user user@domain.com",
+    "responseCode": 500
+}
+```
+
+Lets you create a report, with transactions, in a user’s account from your external system.
```
Comment (yuwenmemon) [inline source/index.html.md]: Lets you create a report, with transactions, in a user’s account.

File: source/index.html.md [hunk 30]
```diff
@@ -1,185 +1,1260 @@
...
+                    }
+                }
+            ]
+        }
+    }'
+```
+
+> Response
+
+> -  A success response message is comprised of a `responseCode` `200`, and the list of transactions that were created. Each transaction also has a unique `transactionID`.
```
Comment (yuwenmemon) [inline source/index.html.md]: NAB: Double space here after the `-`

File: source/index.html.md [hunk 31]
```diff
@@ -1,185 +1,1260 @@
...
+Allows you to create expenses in a user’s account.
+
+### `requestJobDescription` format
+
+- `inputSettings`
+
+Name | Format | Valid values | Description
+-------- | --------- | ---------------- | ---------
+type | String | "expenses" | Specifies to the job that it has to create expenses.
+employeeEmail | String | A valid email address | The expenses will be created in that account. |
```
Comment (yuwenmemon) [inline source/index.html.md]: The expense will be created in this account.

File: source/index.html.md [hunk 32]
```diff
@@ -1,185 +1,1260 @@
...
+transactionList | JSON array | See below | List of expense objects. |
+
+- `expense` objects
+
+Name | Format | Description
+-------- | --------- | ---------------- | ---------
+merchant | String | The name of the merchant of the expense. |
```
Comment (yuwenmemon) [inline source/index.html.md]: The name of the expense's merchant.

File: source/index.html.md [hunk 33]
```diff
@@ -1,185 +1,1260 @@
...
+
+Lets you get a list of policies with some relevant information about them. Optionally, you may retrieve policies for another user on a domain you have been granted access to.
+
+- `inputSettings`
+
+Name | Format | Valid values | Description
+-------- | --------- | ---------------- | ---------
+type | String | "policyList" | Specifies to the job that it has to get a policy summary list.
+**Optional elements** |
+adminOnly | Boolean | true, false| Whether or not to get only policies for which the user is an admin for |
```
Comment (yuwenmemon) [inline source/index.html.md]: Whether or not to only get policies for which the user is an admin for.

File: source/index.html.md [hunk 34]
```diff
@@ -1,185 +1,1260 @@
...
+- `requestJobDescription`
+
+Name | Format | Valid values | Description
+-------- | --------- | ---------------- | ---------
+**Optional elements** |
+categories | JSON object | See `categories` below | Replace or update the existing tags of the policy with the ones provided.
+tags | JSON object | See the dedicated [Update tags](#update-tags) section below | Replace the existing tags of the policy with the ones provided.
+reportFields | JSON object | See `reportFields` below | Replace or update the existing report fields of the policy with the ones provided.
+
+- `inputSettings`
```
Comment (yuwenmemon) [inline source/index.html.md:947]: Replace or update the existing categories of the policy with the ones provided.

File: source/index.html.md [hunk 35]
```diff
@@ -1,185 +1,1260 @@
...
+
+- `categories`
+
+Name | Format | Valid values | Description
+-------- | --------- | ---------------- | ---------
+action | String | "merge", "replace" | Specifies how the categories will be updated. <br>- "replace" removes all existing categories and replaces them with the specified list<br>- "merge" keeps existing categories and updates/adds the ones specified.
+data | JSON array | See below | |
+
+- `categories.data` objects
+
```
Comment (yuwenmemon) [inline source/index.html.md:962]: nab: close the br tags (`<br/>`)

Comment (yuwenmemon) [reply source/index.html.md:962]: Theres a few repeat cases above and below, up to you

File: source/index.html.md [hunk 36]
```diff
@@ -1,185 +1,1260 @@
...
+    "responseMessage": "Column EmployeeEmail not found in CSV header",
+    "responseCode": 500
+}
+```
+
+Lets you manage employees on an Expensify policy. Use this to configure or provision accounts for new hires, as well as to update existing employee data.
+
+The employee data must be passed in the request parameter `data`.
+
+- `inputSettings`
```
Comment (yuwenmemon) [inline source/index.html.md:1115]: Use this to configure or provision accounts for new hires, as well as update existing employee data.

File: source/index.html.md [hunk 37]
```diff
@@ -1,185 +1,1260 @@
...
+**Optional CSV columns** |
+EmployeeUserId | String | | The User ID of the employee.
+EmployeePayrollId | String | | The Payroll ID of the employee.
+ForwardManagerEmail | String | Any valid email address or an empty value | To whom reports will be sent to after the manager clicks 'Approve and Forward'.
+
+Optional attributes can be left blank. In that case, the corresponding value will be removed for existing employees, and left blank for new employees.
+
+## Expense rules updater
+
+```shell
```
Comment (yuwenmemon) [inline source/index.html.md:1142]: Optional attributes can be provided as a blank string.

Comment (francoisl) [reply source/index.html.md:1142]: No because they don't even need to be provided

**Comment by francoisl** [comment]
Updated

File: source/index.html.md (hunk 37 above)
Comment (yuwenmemon) [reply source/index.html.md:1142]: If optional attributes are not provided or left blank, the corresponding value will be removed for existing employees, and left blank for new employees.

How about that?

**Comment by francoisl** [comment]
Updated

**Comment by yuwenmemon** [comment]
💥

**Comment by shawnborton** [comment]
wowee

**Pull request merged**